cicecore/cicedyn/infrastructure/io/io_pio2/ice_history_write.F90

```diff
@@ -232,64 +237,40 @@ subroutine ice_write_hist (ns)
       ! define coordinate variables:  time, time_bounds
       !-----------------------------------------------------------------
 
-      call ice_pio_check(pio_def_var(File,'time',pio_double,(/timid/),varid), &
-           subname//' ERROR: defining var time',file=__FILE__,line=__LINE__)
-      call ice_pio_check(pio_put_att(File,varid,'long_name','time'), &
-           subname//' ERROR: defining att long_name time',file=__FILE__,line=__LINE__)
-
       write(cdate,'(i8.8)') idate0
-      write(title,'(a,a4,a1,a2,a1,a2,a1,i2.2,a1,i2.2,a1,i2.2)') 'days since ', &
+      write(cal_units,'(a,a4,a1,a2,a1,a2,a1,i2.2,a1,i2.2,a1,i2.2)') 'days since ', &
             cdate(1:4),'-',cdate(5:6),'-',cdate(7:8),' ', &
             hh_init,':',mm_init,':',ss_init
-      call ice_pio_check(pio_put_att(File,varid,'units',trim(title)), &
-           subname//' ERROR: defining att units '//trim(title),file=__FILE__,line=__LINE__)
 
       if (days_per_year == 360) then
-         call ice_pio_check(pio_put_att(File,varid,'calendar','360_day'), &
-              subname//' ERROR: defining att calendar 360',file=__FILE__,line=__LINE__)
+         cal_att='360_day'
       elseif (days_per_year == 365 .and. .not.use_leap_years ) then
-         call ice_pio_check(pio_put_att(File,varid,'calendar','noleap'), &
-              subname//' ERROR: defining att calendar noleap',file=__FILE__,line=__LINE__)
+         cal_att='noleap'
       elseif (use_leap_years) then
-         call ice_pio_check(pio_put_att(File,varid,'calendar','Gregorian'), &
-              subname//' ERROR: defining att calendar Gregorian',file=__FILE__,line=__LINE__)
+         cal_att='Gregorian'
       else
          call abort_ice(subname//' ERROR: invalid calendar settings')
       endif
 
+      time_coord = coord_attributes('time', 'time', trim(cal_units))
+      call ice_hist_coord_def(File, time_coord, pio_double, (/timid/), varid) !why is pio_double this not lprecision ?
```

Comment:
@apcraig Do you know why the time coord is always hardcoded to be type double? (i.e. 1 second as a fraction of a day is still much bigger than the minimum a type single would support?)

Reply:
Ahh... Time.....

The problem is not the precision per se, it's having the digits to support accuracy overall.  There are a lot of seconds in 2024 years, for instance, 11 digits worth to be exact.  Single precision just isn't enough.  And it's not just when we count in seconds, even if you count days, you still need the precision to have 1/86400th of a day if we want to resolve time to the second.  We're sort of lucky we only need doubles to count to 1000s of years.  If we started to run millions of years (or needed to resolve milliseconds), we might be in trouble.  There are ways around this, like creating offsets and not counting from zero, and we do that.  But in general, you don't want to end up in a situation where time accuracy is lost, so double is safer.

Reply:
Ok - maybe just remove that comment in the code next time you are doing a tidy up then :)